packages/vite/src/node/server/ws.ts

```diff
@@ -30,7 +30,9 @@ export function createWebSocketServer(
   const hmr = isObject(config.server.hmr) && config.server.hmr
   const wsServer =
     (hmr && hmr.server) ||
-    ((!(hmr && hmr.port) || hmr.port !== config.server.port) && server)
+    ((!(hmr && hmr.port) ||
```

Comment:
should this conditional be broken apart / documented?  This looks very complex and maybe thats the reason the first fix attempt didn't take every case into account.

Reply:
done